bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerMetadata.java

```diff
@@ -422,6 +427,18 @@ public static LedgerMetadata parseConfig(byte[] bytes, Version version, Optional
         } else {
             lc.ackQuorumSize = lc.writeQuorumSize;
         }
+        if (data.hasLedgerType()) {
+            switch (data.getLedgerType()) {
+                case VD_JOURNAL:
+                   lc.ledgerType = LedgerType.VD_JOURNAL;
+                   break;
+                case PD_JOURNAL:
+                   lc.ledgerType = LedgerType.PD_JOURNAL;
+                   break;
+                default:
+                    throw new IllegalStateException("illegal ledgerType "+data.getLedgerType()+" is metadata");
```

Comment:
please throw IOException instead of IllegalStateException. because this means the metadata is corrupted or unknown.